boot-speed/clouds/measure-cloud.py

```diff
@@ -23,6 +24,75 @@
 known_clouds = ['ec2', 'gce']
 
 
+class EC2Instspec:
+    def __init__(self, *, release, inst_type, region, ec2_subnetid, ec2_sgid):
+        # Defaults
+        self.region = "us-east-1"
```

Comment:
These could just be the keyword argument defaults, I think?

(Except for `sgid`, because using mutable objects as defaults is an anti-pattern: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html)

Reply:
I did it like this because the keyword arguments are all getting passed to `__init__`, even if they are `None`, and thus the keyword defaults are always getting overwritten. And I can't set the defaults as the `argparse` arguments default values, as I need different values for different clouds.

But I may be missing the right way to do this, and I'd be happy to, as I don't really like how this is currently implemented, so suggestions are welcome!

Reply:
Ah, hm, that is tricky. You _could_ construct a kwarg dict dynamically in the call site:

```py
kwargs = {}
if args.release:
    kwargs['release'] = args.release
...
EC2Instspec(**kwargs)
```
but that's really just moving the args handling from one place to another, which doesn't gain us a huge amount.

I think we can just add a comment with what you explained above and move on. :-)